src/mscorlib/shared/System/IO/Path.Unix.cs

```diff
@@ -44,91 +44,21 @@ public static string GetFullPath(string path)
             return result;
         }
 
-        /// <summary>
-        /// Try to remove relative segments from the given path (without combining with a root).
-        /// </summary>
-        /// <param name="skip">Skip the specified number of characters before evaluating.</param>
-        private static string RemoveRelativeSegments(string path, int skip = 0)
+        public static string GetFullPath(string path, string basePath)
         {
-            bool flippedSeparator = false;
+            if (basePath == null)
```

Comment:
Sorry, missed that you aren't checking path for null here- should check that as well.  Check it before basePath.

Reply:
Can you review the tests for both the PR ?